src/frontend/qt_sdl/RAMInfoDialog.cpp

```diff
@@ -0,0 +1,296 @@
+
+/**
+ * RAMSearchThread 
+ */
+
+RAMSearchThread::RAMSearchThread(RAMInfoDialog* dialog) : Dialog(dialog)
+{
+    RowDataVector = new std::vector<ramInfo_RowData>();
+}
+
+RAMSearchThread::~RAMSearchThread()
+{
+    Stop();
+    if (RowDataVector)
+    {
+        delete RowDataVector;
+        RowDataVector = nullptr;
+    }
+}
+
+void RAMSearchThread::Start(const s32& searchValue, const ramInfoSTh_SearchMode& searchMode)
+{
+    SearchValue = searchValue;
+    SearchMode = searchMode;
+    start();
+}
+
+void RAMSearchThread::Start(const ramInfoSTh_SearchMode& searchMode)
+{
+    SearchMode = searchMode;
+    start();
+}
+
+void RAMSearchThread::Stop()
+{
+    SearchRunning = false;
+    RowDataVector->clear();
+    quit();
+    wait();
+}
+
+void RAMSearchThread::run()
+{
+    SearchRunning = true;
+    u32 progress = 0;
+
+    // Pause game running
+    emuThread->emuPause();
+
+    // For following search modes below, RowDataVector must be filled.
+    if (SearchMode == ramInfoSTh_SearchAll || RowDataVector->size() == 0)
+    {
+        // First search mode
+        for (u32 addr = 0x02000000; SearchRunning && addr < 0x02000000+NDS::MainRAMMaxSize; addr += SearchByteType)
+        {
+            const s32& value = GetMainRAMVAlue(addr, SearchByteType);
+
+            RowDataVector->push_back({ addr, value, value });
+
+            // A solution to prevent to call too many slot.
+            u32 newProgress = (int)((addr-0x02000000) / (NDS::MainRAMMaxSize-1.0f) * 100);
+            if (progress < newProgress)
+            {
+                progress = newProgress;
+                emit SetProgressbarValue(progress);
+            }
+        }
+    }
+    
+    if (SearchMode == ramInfoSTh_Default)
+    {
+        // Next search mode
+        std::vector<ramInfo_RowData>* newRowDataVector = new std::vector<ramInfo_RowData>();
+        for (u32 row = 0; SearchRunning && row < RowDataVector->size(); row++)
+        {
+            const u32& addr = RowDataVector->at(row).Address;
+            const s32& value = GetMainRAMVAlue(addr, SearchByteType);
+
+            if (SearchValue == value)
+                newRowDataVector->push_back({ addr, value, value });
+
+            // A solution to prevent to call too many slot.
+            u32 newProgress = (int)(row / (RowDataVector->size()-1.0f) * 100);
+            if (progress < newProgress)
+            {
+                progress = newProgress;
+                emit SetProgressbarValue(progress);
+            }
+        }
+        delete RowDataVector;
+        RowDataVector = newRowDataVector;
+    }
+
+    // Unpause game running
+    emuThread->emuUnpause();
+
+    SearchRunning = false;
+}
+
+void RAMSearchThread::SetSearchByteType(const ramInfo_ByteType& bytetype)
+{
+    SearchByteType = bytetype;
+}
+
+ramInfo_ByteType RAMSearchThread::GetSearchByteType() const
+{
+    return SearchByteType;
+}
+
+std::vector<ramInfo_RowData>* RAMSearchThread::GetResults()
+{
+    return RowDataVector;
+}
```

Comment:
could you add a newline to the end of the file?
```suggestion
}

```

Reply:
Sure! (sorry for replying late)